File: Consultorio.API/Consultorio.API/Model/Consulta.cs
﻿using Consultorio.API.Model.Enum;
using System.ComponentModel.DataAnnotations.Schema;

namespace Consultorio.API.Model
{
    public class Consulta : Entity
    {
        public TipoConsulta TipoConsulta { get; set; }
        [Column(TypeName = "text")]
        public string? Descricao { get; set; }
        public DateTime Data { get; set; }
        public int MedicoId { get; set; }
        public Medico Medico { get; set; }
        [Column(TypeName = "varchar(7)")]
        public string MedicoCRM { get; set; }
        public int PacienteId { get; set; }
        public Paciente Paciente { get; set; }
        public ICollection<SintomaConsulta>? Sintomas { get; set; }
    }
}

File: Consultorio.App/src/app/pages/consultas/model/consulta.ts
import { Genero } from '../../../enum/Genero.enum';
import { TipoConsulta } from '../../../enum/TipoConsulta.enum';
import { TipoSanguineo } from '../../../enum/TipoSanguineo.enum';
import { SintomaConsulta } from './sintomaConsulta';

export interface Consulta {
  id: number;
  tipoConsulta: TipoConsulta;
  descricao: string;
  data: Date;
  medicoId: number;
  medicoCRM: string;
  especializacao: string;
  medicoNome: string;
  medicoNomeSocial: string;
  pacienteId: number;
  nome: string;
  nomeSocial: string;
  cpf: string;
  dataNascimento: Date;
  email: string;
  telefone: string;
  tipoSanguineo: TipoSanguineo;
  genero: Genero;
  sintomas?: SintomaConsulta[];
}
